pkg/kubelet/kubelet.go

```diff
@@ -932,12 +929,20 @@ func (kl *Kubelet) pullImage(img string, ref *api.ObjectReference) error {
 
 // Kill all running containers in a pod (includes the pod infra container).
 func (kl *Kubelet) killPod(runningPod kubecontainer.Pod) error {
+
 	// Send the kills in parallel since they may take a long time.
 	errs := make(chan error, len(runningPod.Containers))
 	wg := sync.WaitGroup{}
 	for _, container := range runningPod.Containers {
 		wg.Add(1)
 		go func(container *kubecontainer.Container) {
+			// Call the networking plugin for teardown.
```

Comment:
A couple thoughts:

First, there is no validation to prevent a user from creating a container with the reserved name.  Should we add one?

Second, this is one of the many call sites where it would be really advantageous to have a convenient way of marking a container as being a system artifact, ie, not present in user config/payloads.  Thinking back to another life time, I remembered that the Java reflection API has a notion called 'synthetic' in the reflection API - a synthetic method is one that is not present in user code but is generated by the compiler -- for example, implied no-arg constructors, serialization constructors, constructors to nested classes, etc.  I think this at some level is the same idea.  So, what about this:

1.  Add a 'Synthetic' field to `Container`
2.  Make it so that a user cannot set the value of this field; ie, it is ignored on create / update in the rest API.
3.  Kubelet or future container runtime can use this field to mark synthetic containers it creates (ie, pod infra container)

Any thoughts @yifan-gu @vmarmol @dchen1107 @thockin ?